### PR TITLE
Implement `numerator` & `denominator` on Float & Numeric

### PR DESCRIPTION
**Pull request by stevegeek**
Implement `numerator` & `denominator` on Float & Numeric

Also simplify `is_finite` to just use `is_infinity`

Adds specs for the Numeric `denominator` and `numerator` methods (I should upstream these specs to `ruby/spec`)

### CODE REVIEW

File: spec/core/numeric/denominator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  it "calls self#to_r then returns the #denominator" do
+    result = mock("Numeric#to_r result")
+    result.should_receive(:denominator).and_return(12)
+    @obj.should_receive(:to_r).and_return(result)
+
+    @obj.denominator.should == 12
+  end
+end
```
Comment (seven1m) [inline spec/core/numeric/denominator_spec.rb:16]: If you do indeed plan to PR these into ruby/spec upstream, then this is fine. However, our usual technique is to add our tests in the `test/natalie` directory, e.g. `test/natalie/float_test.rb` or similar.

**Review by seven1m — APPROVED** [review]
Let me know if you want to make changes before I merge. Looks good to me!

File: spec/core/numeric/denominator_spec.rb (hunk 1 above)
Comment (stevegeek) [reply spec/core/numeric/denominator_spec.rb:16]: I was planning to, but then I just noticed they already exist 🤦 

https://github.com/ruby/spec/blob/master/core/numeric/denominator_spec.rb

I will update PR to add them

Comment (seven1m) [reply spec/core/numeric/denominator_spec.rb:16]: Ahh yeah, we just copy over files as-needed. Thanks!

**Pull request merged**